### ISSUE TITLE
[QUESTION] Why `PinoLogger` instance don't show the error object

### ISSUE THREAD

**Issue by micalevisk** [iamolegga/nestjs-pino#304]
When I use a `PinoLogger` instance to display some error object like `this.logger.error(Error('foo'))` I get this:
```json
{
  "level": 50,
  "time": 1590422389076,
  "pid": 27787,
  "hostname": "--",
  "name": "app",
  "req": {
    "id": 1,
    "method": "GET",
    "url": "--",
    "headers": {
      "host": "localhost:3000",
      "user-agent": "insomnia/7.1.1",
      "accept": "*/*"
    },
    "remoteAddress": "127.0.0.1",
    "remotePort": 52126
  },
  "context": "ExceptionsHandler"
}
```
seems like the arg isn't serialized by `pino.stdSerializers.err` because 
```js
this.logger.error( pino.stdSerializers.err(Error('foo')) )
```
works as expected (with `{ type, message, stack }`). This happen even when I explicity define the serializers. Is it the right behavior?

Great lib btw!

**Comment by iamolegga** [comment]
Hi, it's interesting, could you please provide minimal example repo on this?

**Comment by micalevisk** [comment]
While testing the minimal example I figured out that this issue appear when I define the context to the `PinoLogger` instance. Without calling `logger.setContext` or using the `@InjectPinoLogger` decorator, `this.logger.error(Error('foo'))` show the error object.

You can download the example here: https://codesandbox.io/s/nestjs-pino-issue-304-lgxwv

**Comment by micalevisk** [comment]
I just realize this is a bug :/

This happen due to the following code
https://github.com/iamolegga/nestjs-pino/blob/141a69bea58516d33d871ce4ea715ca108f4444b/src/PinoLogger.ts#L92
because in my case `firstArg` has non-enumerable props (`stack` and `message`) so they aren't copied to the new object created there.

**Comment by iamolegga** [comment]
Good catch, thanks. I'll try to see what can be done here

**Comment by kirit0s** [comment]
Pino expects Error object in `err` field and can correctly serialize it.

Example: 
```js
this.logger.error({ err: new Error('message') })
```

**Comment by iamolegga** [comment]
Added extra check if passing `instance of Error`, and if so transfering it to `err` field. Anyway that's more related to `pino` than to this lib. if you passing some custom error please check this out: https://github.com/Microsoft/TypeScript-wiki/blob/master/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work. Also this issue might be useful too: https://github.com/pinojs/pino/issues/579

shipped in v1.3.0
